### PR TITLE
[Fix] Matrix Type Representation

### PR DESCRIPTION
**Pull request by jjfumero**
#### Description

As reported in issue https://github.com/beehive-lab/TornadoVM/discussions/213, TornadoVM produces wrong results for some of the matrix types. THe problem was due to different memory representation of matrix types. This PR fixes the issue by providing a common representation (Rows X Columns). 

#### Backend/s tested

Mark the backends affected by this PR.

- [X] OpenCL
- [X] PTX
- [X] SPIRV

#### OS tested

Mark the OS where this PR is tested.

- [X] Linux
- [ ] OSx
- [ ] Windows

#### Did you check on FPGAs?

If it is applicable, check your changes on FPGAs.

- [ ] Yes
- [X] No

#### How to test the new patch?

```bash
make 
make tests
make BACKEND=spirv
make tests
make BACKEND=ptx
make tests
```

Also passing KFusion and Ray-Tracer

### CODE REVIEW

File: tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/Matrix2DFloat.java [hunk 1]
```diff
@@ -109,39 +110,39 @@ public float[] getFlattenedArray() {
...
-        storage[toRowMajor(i, j, N)] = value;
+        storage[toRowMajor(i, j, COLUMNS)] = value;
     }
 
     public int M() {
-        return M;
+        return ROWS;
     }
 
     public int N() {
```
Comment (stratika) [inline tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/Matrix2DFloat.java:124]: Refactor the name of methods `M()`, `N()`.

File: tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/Matrix2DFloat4.java [hunk 2]
```diff
@@ -74,73 +74,73 @@ public class Matrix2DFloat4 implements PrimitiveStorage<FloatBuffer> {
...
 
     public int M() {
-        return M;
+        return ROWS;
     }
 
     public int N() {
-        return N;
+        return COLUMNS;
     }
```
Comment (stratika) [inline tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/Matrix2DFloat4.java:124]: Refactor the name of methods `M()`, `N()`.

File: tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/Matrix2DInt.java [hunk 3]
```diff
@@ -103,39 +103,39 @@ public int[] getFlattenedArray() {
...
 
     public int M() {
-        return M;
+        return ROWS;
     }
 
     public int N() {
-        return N;
+        return COLUMNS;
     }
```
Comment (stratika) [inline tornado-api/src/main/java/uk/ac/manchester/tornado/api/collections/types/Matrix2DInt.java:119]: Refactor the name of methods `M()`, `N()`.

**Comment by jjfumero** [comment]
I would separate the refactoring from the fix.

**Comment by stratika** [comment]
> I would separate the refactoring from the fix.

But, this PR seems to apply refactoring too. `M -> ROWS`, `N -> COLUMNS`

**Comment by jjfumero** [comment]
> > I would separate the refactoring from the fix.
> 
> But, this PR seems to apply refactoring too. `M -> ROWS`, `N -> COLUMNS`

Yes, but this will break KFusion and other use cases. I am fine with the change, but we should update all satellite repos accordingly as well.

**Review by mairooni — APPROVED** [review]

**Comment by jjfumero** [comment]
@stratika , I implemented the changes. I also adapted KFusion

**Comment by jjfumero** [comment]
Kfusion is now passing as well.

**Review by stratika — APPROVED** [review]
All unit-tests have been tested for all backends on my setup.

**Pull request merged**